install

```diff
@@ -145,11 +147,13 @@ def git
 
   return unless @git
 
-  # Github only supports HTTPS fetches on 1.7.10 or later:
-  # https://help.github.com/articles/https-cloning-errors
-  `#{@git} --version` =~ /git version (\d\.\d+\.\d+)/
-  return if Regexp.last_match(1).nil?
-  return if Version.new(Regexp.last_match(1)) < "1.7.10"
+  if can_use_https?
+    # GitHub only supports HTTPS fetches on 1.7.10 or later:
+    # https://help.github.com/articles/https-cloning-errors
+    `#{@git} --version` =~ /git version (\d\.\d+\.\d+)/
+    return if Regexp.last_match(1).nil?
+    return if Version.new(Regexp.last_match(1)) < "1.7.10"
```

Comment:
Seems a bit weird to return `nil` here if there's a valid Git but we're not going to do HTTPS cloning. Could perhaps assume given the Git versions that if `can_use_https` is fine then the Git version is too or add another function `can_git_use_https?`?